cmd/skopeo/sync.go

```diff
@@ -417,6 +418,15 @@ func imagesToCopyFromRegistry(registryName string, cfg registrySyncConfig, sourc
 // encountered.
 func imagesToCopy(source string, transport string, sourceCtx *types.SystemContext) ([]repoDescriptor, error) {
 	var descriptors []repoDescriptor
+	defer func() {
```

Comment:
(I also don’t think using `defer` is helpful for understanding the code structure in this case, but that’s a side point when the whole idea currently seems problematic.)